pex/resolver.py

```diff
@@ -347,32 +376,46 @@ def from_request(cls, build_request, dist_root):
         )
         return cls(request=build_request, atomic_dir=AtomicDirectory(dist_dir))
 
+    def __init__(
+        self,
+        request,  # type: BuildRequest
+        atomic_dir,  # type: AtomicDirectory
+    ):
+        # type: (...) -> None
+        self.request = request
+        self._atomic_dir = atomic_dir
+
     @property
     def is_built(self):
-        return self.atomic_dir.is_finalized
+        # type: () -> bool
+        return self._atomic_dir.is_finalized
 
     @property
     def build_dir(self):
-        return self.atomic_dir.work_dir
+        # type: () -> str
+        return self._atomic_dir.work_dir
 
     @property
     def dist_dir(self):
-        return self.atomic_dir.target_dir
+        # type: () -> str
+        return self._atomic_dir.target_dir
 
     def finalize_build(self):
-        self.atomic_dir.finalize()
+        # type: () -> Iterator[InstallRequest]
+        self._atomic_dir.finalize()
         for wheel in os.listdir(self.dist_dir):
             yield InstallRequest.create(self.request.target, os.path.join(self.dist_dir, wheel))
 
 
-class InstallRequest(namedtuple("InstallRequest", ["target", "wheel_path", "fingerprint"])):
+class InstallRequest(object):
```

Comment:
Does this need `__eq__` and `__hash__`? May be worth doing `__repr__` too.

Reply:
Ditto on the other classes in this file that don't use namedtuple anymore.

Reply:
They do not need `__eq__` / `__hash__` but `{Build,Install}Request` could both stand a `__repr__` for the verbose logs.